bluesky_widgets/qt/run_tree.py

```diff
@@ -0,0 +1,195 @@
+from qtpy import QtCore
+from qtpy.QtCore import (
+    QAbstractItemModel,
+    QModelIndex,
+    QTimer,
+    Qt,
+)
+from qtpy.QtWidgets import QAbstractItemView, QHeaderView, QTreeView
+
+from databroker import catalog
+from databroker.core import BlueskyEventStream
+
+class RunTree(object):
+    """Lazily populate the tree as data is requested. """
+    def __init__(self, run):
+        self.run = run
+        self.children = []
+
+        uid = RunNode(self.run, 'uid', self.run.metadata['start']['uid'], None, self)
+        start = RunNode(self.run, 'start', 'dict', self.run.metadata['start'], self)
+        start.num_children = len(self.run.metadata['start'])
+        stop = RunNode(self.run, 'stop', 'dict', self.run.metadata['stop'], self)
+        stop.num_children = len(self.run.metadata['stop'])
+        streams = RunNode(self.run, 'streams', 'events (' + str(len(self.run)) + ')', None, self)
```

Comment:
I would nudge you toward embracing f-strings over `+` for this kind of construction because they are more readable and performant --- i.e. `f"events ({len(self.run)})"`.

Separately, I guess this is really number of _streams_ so the word "events" might be unhelpful here. Maybe we just want `f"({len(self.run)})"`.

Reply:
I appreciate the nudge, completely agree, and it is much shorter/more readable.